File: src/codes/Users_Manual/stm/focus_stm.rs
# use autd3::prelude::*;
# fn main() {
let center = Point3::new(0., 0., 150.0 * mm);
let point_num = 200;
let radius = 30.0 * mm;
# let _ = 
FociSTM {
    foci: (0..point_num)
        .map(|i| {
            let theta = 2.0 * PI * i as f32 / point_num as f32;
            let p = radius * Vector3::new(theta.cos(), theta.sin(), 0.0);
            center + p
        })
        .collect::<Vec<_>>(),
    config: 1.0 * Hz,
};
# }

File: src/codes/Users_Manual/stm/focus_stm.cpp
//~#include <ranges>
//~#include<autd3.hpp>
//~#include<autd3/link/nop.hpp>
//~using namespace std::ranges::views;
//~int main() {
//~using namespace autd3;
const Point3 center(0, 0, 150);
constexpr auto points_num = 200;
constexpr auto radius = 30.0f;
FociSTM(iota(0) | take(points_num) | transform([&](auto i) {
          const auto theta = 2.0f * pi * static_cast<float>(i) /
                             static_cast<float>(points_num);
          Point3 p =
              center + radius * Vector3(std::cos(theta), std::sin(theta), 0);
          return p;
        }) | std::ranges::to<std::vector<Point3>>(),
        1.0f * Hz);
//~return 0; }